src/main/java/in/org/projecteka/hiu/consent/GrantedConsentTask.java

```diff
@@ -51,10 +52,11 @@ public GrantedConsentTask(ConsentRepository consentRepository,
                 .switchIfEmpty(Mono.error(ClientError.consentRequestNotFound()))
                 .flatMap(consentRequest -> {
                     var cmSuffix = getCmSuffix(consentRequest.getPatient().getId());
-                    return Flux.fromIterable(consentNotification.getConsentArtefacts())
-                            .flatMap(reference -> perform(reference, consentNotification.getConsentRequestId(),
-                                    cmSuffix))
-                            .then();
+                    return consentRepository.updateConsentRequestStatus(
+                            ConsentStatus.GRANTED, consentNotification.getConsentRequestId())
+                            .then(Flux.fromIterable(consentNotification.getConsentArtefacts())
+                                    .flatMap(reference -> perform(reference, consentNotification.getConsentRequestId(), cmSuffix))
+                                    .then());
```

Comment:
Why is this needed.

Reply:
we were not updating the consent_request status in case of GRANT.